xarray/backends/plugins.py

```diff
@@ -81,7 +81,10 @@ def sort_backends(backend_entrypoints):
 
 
 def build_engines(pkg_entrypoints):
-    backend_entrypoints = BACKEND_ENTRYPOINTS.copy()
+    backend_entrypoints = {}
+    for backend_name, backend in BACKEND_ENTRYPOINTS.items():
+        if backend.are_dependencies_installed():
+            backend_entrypoints[backend_name] = backend
```

Comment:
```suggestion
    backend_entrypoints = {
        backend_name: backend
        for backend_name, backend in BACKEND_ENTRYPOINTS.items()
        if backend.are_dependencies_installed()
    }
```

Reply:
I prefer avoiding comprehension if they are complex since they are less readable.

Reply:
In this blackened format I think it's as readable as the normal loop. But I don't feel that strongly about this, if you prefer the other one that's fine too.